Diagnostics/HealthChecker/Analyzer/Security/Invoke-AnalyzerSecuritySettings.ps1

```diff
@@ -293,26 +293,58 @@ function Invoke-AnalyzerSecuritySettings {
+    if ($null -ne $HealthServerObject.ExchangeInformation.BuildInformation.FIPFSUpdateIssue) {
+        $fipfsInfoObject = $HealthServerObject.ExchangeInformation.BuildInformation.FIPFSUpdateIssue
+        $highestVersion = $fipfsInfoObject.HighesVersionNumberDetected
+        $fipfsIssueBaseParams = @{
             Name             = "FIP-FS Update Issue Detected"
             Details          = $true
             DisplayWriteType = "Red"
         }
-        Add-AnalyzedResultInformation @params
+        $moreInformation = "More Information: https://aka.ms/HC-FIPFSUpdateIssue"
+
+        if ($fipfsInfoObject.ServerRoleAffected -eq $false) {
+            # Server role is not affected by the FIP-FS issue so we don't need to check for the other conditions.
+            Write-Verbose "The Exchange server runs a role which is not affected by the FIP-FS issue"
+        } elseif (($fipfsInfoObject.FIPFSFixedBuild -eq $false) -and
+            ($fipfsInfoObject.BadVersionNumberDirDetected)) {
+            # Exchange doesn't run a build which is resitent against the problematic pattern
+            # and a folder with the problematic version number was detected on the computer.
+            $params = $baseParams + $fipfsIssueBaseParams
+            Add-AnalyzedResultInformation @params
 
-        $params = $baseParams + @{
-            Details                = "More Information: https://aka.ms/HC-FIPFSUpdateIssue"
-            DisplayWriteType       = "Red"
-            DisplayCustomTabNumber = 2
-        }
-        Add-AnalyzedResultInformation @params
-    } elseif ($null -eq $HealthServerObject.ExchangeInformation.BuildInformation.AffectedByFIPFSUpdateIssue) {
-        $params = $baseParams + @{
-            Name             = "FIP-FS Update Issue Detected"
-            Details          = "Error: Failed to find the scan engines on server, this can cause issues with transport rules as well as the malware agent."
-            DisplayWriteType = "Red"
+            $params = $baseParams + @{
+                Details                = $moreInformation
+                DisplayWriteType       = "Red"
+                DisplayCustomTabNumber = 2
+            }
+            Add-AnalyzedResultInformation @params
+        } elseif (($fipfsInfoObject.FIPFSFixedBuild) -and
+            ($fipfsInfoObject.BadVersionNumberDirDetected)) {
+            # Exchange runs a build that can handle the problematic pattern. However, we found
+            # a high-version folder which should be removed (recommendation).
+            $fipfsIssueBaseParams.DisplayWriteType = "Yellow"
+            $params = $baseParams + $fipfsIssueBaseParams
+            Add-AnalyzedResultInformation @params
+
+            $params = $baseParams + @{
+                Details                = "We've detected the problematic FIP-FS version: $highestVersion`r`n`t`tAlthough it should not cause any problems, we recommend performing a FIP-FS reset.`r`n`t`t$moreInformation"
```

Comment:
Since this based off the folder information, maybe we should provide that vs the version that is truly running?

I think that is part of the problem, `Get-EngineUpdateInformation` is providing a different version than the latest directory version possible.  

Maybe just change the wording to be "problematic FIP-FS version folder detected".